### PR TITLE
Replaces Free Guns From Sci With Parts Kits

### PR DESCRIPTION
**Pull request by Bm0n**
## What Does This PR Do
Ports most of https://github.com/tgstation/tgstation/pull/56133

Replaces lock boxed guns from science with gun parts kits(still made in the protolathe) which can be crafted into advanced weapons.

All guns from sci need their respective parts kit plus either an energy gun, a laser gun or KPA depending on which weapon is being crafted. Look at the chart in the images section, it's not exactly the same as TG but I think this makes more sense.

The parts kits have tech levels the same as the guns they are crafted into so that they can be used in research.
## Why It's Good For The Game
The arms race between sec and the antag never ends as it stands currently and I personally think one of the sources of the issue is the protolathe and sci guns. This PR will slow and lessen the use and abuse of the protolathe to produce weapons en masse, making it so that you require a pre-existing gun to be able to craft advanced science weapons.

I was very active on TG when this PR was first introduced and it significantly reduced the overall powergaming of sci guns while not making them unobtainium. I saw it as a positive change and shift in the community over there and now wish for this PR to be ported over to Paradise.

I don't think this will affect t-spiders or blob as much as people think it will, both have been tuned down in recent years and I have seen them die repeatedly to stock weapon/welders.
## Images of changes
PR

## Changelog
:cl: Bmon
tweak: Science guns have been replaced by parts kits which can be crafted into their respective weapon using the crafting menu
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
Oooooooh this is gonna be interesting 


I have wanted to cap the level of guns science can shit out in highpop and require atleast some extra work (cargo involvement, which this does), and see how it plays out.


Definitely gonna be testmerging, but I am in favour.

**Comment by datlo** [comment]
Using recipes for this seems unnecessary considering the kit only have a single purpose : Modifying the gun, whereas recipes and crafting are usually used for combining otherwise unrelated items together.

It'd be simpler both to use and maintain to have guns get an attack_by where using a kit on them would modify the gun and consume the kit (which you could get back by an attack_by with a screwdriver on the gun if you wanted to)

**Comment by Bm0n** [comment]
> Using recipes for this seems unnecessary considering the kit only have a single purpose : Modifying the gun, whereas recipes and crafting are usually used for combining otherwise unrelated items together.
> 
> It'd be simpler both to use and maintain to have guns get an attack_by where using a kit on them would modify the gun and consume the kit (which you could get back by an attack_by with a screwdriver on the gun if you wanted to)

I don't know how I'd be able to rope in chemical, 2x tools and cable requirement craft into an attack_by and somehow make it simpler. I think the way TG did it by using the crafting menu makes sense as that's how all other weapons are also made. Making it be an attack_by would remove clarity as well, instead of randomly jabbing tools at a weapon you can see the requirements beforehand.

**Comment by datlo** [comment]
My point was that those requirements are unnecessary, but alright.

**Comment by Qwertytoforty** [comment]
I foresee a huge change in terror spider tactics, with them rushing cargo instead of science now, as instead of 2 places for crew to get weapons, science and cargo, cargo is required for weapons, and crew can not get weapons without them. This does worry me, along with crew having a much longer wait time to get weapons to blob, which is great for things like xray, but less so for AEGs. Blobs often do die to stock and welders if it is roundstart, but much less so for midround. This also means any officer can get an AEG without warden access once armory is opened, by grabbing an egun. I am also curious how this will effect the print rate of rnd weapons that are not AEG / immolator, since someone now has to sacrifice a gun to try it out. Definitely worth a testmerge, but something I do wish to keep an eye on.

**Comment by Coolrune206** [comment]
> > Using recipes for this seems unnecessary considering the kit only have a single purpose : Modifying the gun, whereas recipes and crafting are usually used for combining otherwise unrelated items together.
> > It'd be simpler both to use and maintain to have guns get an attack_by where using a kit on them would modify the gun and consume the kit (which you could get back by an attack_by with a screwdriver on the gun if you wanted to)
> 
> I don't know how I'd be able to rope in a 2x tools, chemical and cable requirement craft into an attack_by and somehow make it simpler. I think the way TG did it by using the crafting menu makes sense as that's how all other weapons are made. Making it be an attack_by would remove clarity as well, instead of randomly jabbing tools at a weapon you can see the requirements beforehand.

Perhaps have the weapon kits for the Ebow and Decloner have a small reagent capacity only accepting radium or mutagen, and only able to be used on the respective weapon if full.

Not sure how I feel about the Lathe ebow being restricted to a mining tool. Can cargo even order KAs outside of the mining conscription kit?

**Comment by Bm0n** [comment]
> Not sure how I feel about the Lathe ebow being restricted to a mining tool. Can cargo even order KAs outside of the mining conscription kit?

It's because Ebows have always been a subtype of KPAs and they operate in the same way. But yes, conscription kits from cargo would be the way to procure one(and it's only 1 KPA per crate so this is good). You need to order laser or eguns from cargo to get the other sci guns so it's no different from the others.

As for the code, I'd rather keep it the same as TG... if anything just for simplicity and to avoid confusion with players who may play one of the many TG downstream.

**Comment by KittyNoodle** [comment]
my only problem with this stems from the fact that this will solidify firepower to sec and make getting weapons as a non-traitor antag even harder

**Comment by procdrone** [comment]
> > Using recipes for this seems unnecessary considering the kit only have a single purpose : Modifying the gun, whereas recipes and crafting are usually used for combining otherwise unrelated items together.
> > It'd be simpler both to use and maintain to have guns get an attack_by where using a kit on them would modify the gun and consume the kit (which you could get back by an attack_by with a screwdriver on the gun if you wanted to)
> 
> I don't know how I'd be able to rope in a 2x tools, chemical and cable requirement craft into an attack_by and somehow make it simpler. I think the way TG did it by using the crafting menu makes sense as that's how all other weapons are made. Making it be an attack_by would remove clarity as well, instead of randomly jabbing tools at a weapon you can see the requirements beforehand.

if you want to include these requirements, maybe a subtype of crafting menu when you use the kit in hand?

**Comment by moxian** [comment]
I am against using crafting menu for this, the same reasons as datlo outlined above. And also that the crafting menu is just  plain clunky to work with.

The only reason to require tools other than the kit and the gun is for immersion / ikea effect.
But that can be achieved without going through crafting menu. One idea, from discord:

> you could also have it so you apply the kit to the gun, making an incomplete AEG / immolator / whatever, then use a screwdriver to assemble it and have the however-long delay happen here

**Comment by Bm0n** [comment]
> When you say 'the legacy way of doing crafting', you're looking at this with the wrong lens.
> 
> This is _not_ crafting, this is gun mods, and we already have a precedent : the syndi silencer, which should be refactored to work into a new gun mod system where this fits in, probably similar to CM, and does not use the crafting system.

You're creating a whole new weapon with different capabilities from a base weapon, a parts kit, tools and cable. It's crafting, not a simple weapons modification like a suppressor which doesn't physically change the capabilities of the weapon.

> 'TG did it this way' is imo not enough of a reason for as long as we remain a separate codebase.

I can literally say the same thing about using attackby as it's Goon's way of doing crafting because they're the only server that I know of crazy enough not to use a crafting menu for stuff like this. CM uses attackby for simple weapon mods which again isn't the same thing.

An attackby crafting recipe from Goon's wiki just to show you the absurdness of it:
image
You wouldn't be able to figure something like this out on your own without first looking at a wiki page which I don't like.

> I mean, would it really? We already have tons of items that use separate crafting menus, like various metal sheets, wood sheets, cloth, leather, etc.

Those menus are for creating rods or girders from a base material like metal, not combining things together.


Sorry for dragging my heels about this, but I genuinely think using the crafting menu is the most logical thing to do considering that we've kept ours up to date with TG's throughout the years.
It's simpler and just works better in the long run for both new and old players alike.

**Comment by Xsingular** [comment]
Please allow me to present an opportunity to discuss the balance implications of this PR, because I can't help but feel that we are glossing over part of what makes this system a necessity on TG


image

In this image, this _single screenspace_, a Warden/HoS/Captain has everything they need to arm the station in case of a crisis.
Top left you can see the armory, south of the clothes vendors you can see the cargo ordering console, and in front of the player character you can see the Security Departmental Techfab, which is connected to the Ore Silo containing the entire output of Mining, and to the Research Servers (which are also automatically synched on TG, unlike Paradise).

As you can see, if TG was to adopt our system, than a single Warden can print dozens, maybe even hundreds of guns unimpeded.


Let's examine the gameplay loop on Paradise:

- On Paradise, you don't have a **Cargo Ordering Console**, so that's a trip Sec has to make. Additionally, the Cargo system in general is way less robust/feature-dense on Para, and attracts less experienced crew than it does on TG, so that's a long trip to make, plus explanation with details to whatever inexperienced Cargo Tech is standing there (if lucky) about how and where to shop these crates.
- On Paradise, you don't have a **Departmental Techfab**, so that's another trip Sec has to make to RnD, our research system is still based on deconstructive analysis, which while faster to max for experienced players, is way harder and obtuse for inexperienced players, making even the required research levels even being there less of a foregone conclusion compared to TG which has the intuitive Click-To-Research TechWebs.
- On Paradise, you don't have an **Ore Silo**, so even should the base guns be available, and the research, you are most likely still going to need to head to the ORM to get more materials (any scientist who ran out of uranium during a blob-caused AEG rush knows that feeling), so that's another potentially massively long trip Sci has to make (or Sec if they feel like helping).

And to put into perspective that non-trivial nature of these trips, please note that on Paradise's most played map, Cyberiad, Research and Cargo are at opposite ends of the southern (aft) half of the station, extremely far from each other.
And then you have Security at the northernmost (fore) point.

All of the aforementioned is a non-issue on TG. 


I like the part kit system a lot more than printing guns (I also like the firing pin system a lot more too for what it matters), but it feels like the system is simply one cog of a larger machine that include Departmental Lathes and Ore Silos and more robust Cargo systems. A simple transplant of the Gun Kits is not likely to yield a positive impact on any loyalist/officer/EoC crewmember gameplay loop, in my humble opinion.

Sorry for making you read all this.


**TL:DR** Massive station-wide threat? 
_TG Warden:_ Come to Sec to arm up boys! We got AEGs, we got Xrays, we got Accelerators, we got NV, we even have Teleshields!
_Para Warden:_ Welp... let me just get my jogging shoes!

**Comment by Bm0n** [comment]
> whatever Xsingular said

Finally, I don't have to argue about attackby crafting

You understand that on TG before parts kits anyone in sec could have printed whatever fully completed sci gun they wanted from the department tech lathe. The limiting factor before parts kits were firing pins, firing pins have now taken a backseat on TG as guns are now are the limiting factor but there was nothing stopping the warden from stockpiling on firing pins before this and having as many guns as they wanted.

I was a very active TG player and I know exactly why all these systems were added and this would work on Paradise despite the differences.

- On TG, the cargo ordering console was added to every department because **each got a round start cargo budget** which departments can spend on supplies. Understand that TG is mostly LRP and that cargo's default budget was almost always spent on junk necessitating independent cargo budgets for each department. This isn't an issue on Paradise for three reasons. One: QMs are actually held accountable on Paradise. Two: Cargo points can be more easily obtained on Paradise. Three: Command tends to work together on Paradise, it'd be as simple as asking the Captain who should be on the bridge to walk over to the cargo ordering console and order said weapons.
- Yes, we don't have departmental lathes on Paradise, the heads are against it and there is nothing I or anyone can do to change their minds, however, this doesn't change anything. You already have to leave brig to get science weapons from Science, nothing has changed. **Techwebs have experiments**- to get advanced weapon development technology to unlock you need to make a **large bomb using toxins** and detonate it in the bomb test range. That is **arguably way harder** than following a list on the wiki to unlock sci guns from RND on Para, especially considering TG uses a more complex toxins system.
- The Ore Silo is a moot point, mainly because Paradise doesn't have department lathes so all the materials end up in Science anyways. Again, nothing has changed from this PR to the current day Paradise.

Before this PR you still needed to leave brig to get weapons from science, the only thing that's changed is that now you need to use weapons from either the armoury or cargo to create them from kits.

Firing pins and departmental lathes will never happen on Paradise because the heads are against it, can't do anything about that. Ore silo I am not sure about but it's not really needed since we don't have department lathes.

All of your points are moot. This PR would work flawlessly for Paradise if not better since people work together on Paradise.

**TL;DR** Massive station-wide threat?
_TG Warden:_Hey guys, I am not going to open the armoury because half of you will probably kill me on the spot and I only have six guns in here that actually fire lethals because cargo refuses to send me more guns- oh, and no one researched RND weapons because that requires toxins knowledge.
_Para Warden:_Wow someone did RND, hey HoS, tell the Captain to order lasers from Cargo, I'll go get kits from RND.

**Comment by AffectedArc07** [comment]
I would take table/menu crafting over attackby crafting for these any day, not just for code complexity reasons, but simplicity for ingame and wiki purposes.

**Comment by Xsingular** [comment]
> Finally, I don't have to argue about attackby crafting.
I have no strong opinion on either position honestly, a lot of the stuff we have is already redundant between the two systems, And while attackby might be more intuitive to most players at this point, whenever I look at the conditional/switch state code it necessitates, I can practically feel the will to live being sapped out of me.
But back to the topic at hand.


> * On TG, the cargo ordering console [...] etc.

Does not change the point that ordering from a readily available console in Brig, with a readily available budget (plus the Cargo chat console, I'd argue) makes it easier for Wardens get what they need and react quickly to the station needs, and even more so for simply just stocking up in advance: it's their budget, they use it as they see fit.
Cargo works closer in tandem with other departments on TG than it does here on Para, exactly because of the points you raised: the fact that points are hard to get, and that the bounty list that Cargo gets always includes stuff from other departments, in turn actively forcing the Cargonians to scout out the other departments and cooperate with them if they want to order any of the memestuff they want, this encourages them to be proactive in the services they provide.

A similar comparison to be found in our userbase is in how a responsible RD would dismiss any request for a Chem Dispenser from non-Chemists, and yet it has become the defacto tradition for Science to give a Chem Dispenser to Hydroponics, simply because Science itself needs help in return from Botany to max their research levels.
This mutual need is what drives closer cooperation and teamwork between the two departments, and that's how it works for Cargo in TG, except it's with every other department.
Paradise's Cargo is mostly independent, in that while some might rarely venture out to get disks from RnD (the only interaction worth the time for most Cargo Techs) most are just content in sitting there and letting the Plasma sheets that mining brings do the heavy lifting as far as points go.
As for QMs being accountable, you cannot hold accountable an inexperienced player, and inexperinced players are the norm for Cargo on Para. QMs are not heads and thus the server rules regarding heads and their competence, or the hour requirement both do not apply. More recent example: Cargo **still** orders riot shotguns against threats that require lethal force when Gamma isn't even in effect. 

> * Yes, we don't have departmental lathes on Paradise, [...]  **Techwebs have experiments**- to get advanced weapon development technology[...] etc.
Fun fact, I took this screenshot about 20 minutes in on my own personal server, and that includes the time it took for me to set up the SM, and the scrubber CO2 atmos pipenet for the coolers in the ordnance lab to work, since I am unfamiliar with the TG admin tools.
I could have even just published a Tier 1 "High Yield Explosives" experiment, which can be done using almost exactly the same steps the simplistic Toxins in Paradise has, but I actually took the time to make a Tritium/O2 mix for a Tier 3 paper instead.
This is literally just one person doing Toxins and then leaving the billion inexperienced Scientists to press buttons in front of a screen. 
Or alternatively, doing it like a passive side job that does not require high engagement, while the scientist is busy with other stuff.


> * The Ore Silo is a moot point,... etc
And yet, RnD runs out of materials all the time, all it takes is for ONE mineral to be in shortage in the lathe to require you to run the entire station's length for some Titanium, Silver, Uranium, or some other silly metal you never thought you'd reasonably run out of, this risk of running out is simply not present for TG players.

> Before this PR you still needed to leave brig to get weapons from science, the only thing that's changed is that now you need to use weapons from either the armoury or cargo to create them from kits.
Before this PR, the standard procedure was to send Armory access, either in-person, or on a timed Guest Pass, officers would come to RnD and do all their shopping there (i.e. "can I have an AEG, an Xray, your best cell, a teleshield and NV SecHuds?")
Another thing this common technique granted was that RnD as well as Cargo and Sec can now arm people in case of a stationwide threat like a Blob or Terrors, the aforementioned distance between these three points actually works in _favor_ of the crew since now they are always close to some part of the station that can offer them a chance at defending their lives.


> Firing pins and departmental lathes will never happen on Paradise because the heads are against it, can't do anything about that. Ore silo I am not sure about but it's not really needed since we don't have department lathes.
Not asking you to change that, I know heads are against it, that was one point I wanted to raise concerning the new TG-ported Cerberon layout, is that it does not take the fact that the awkwardly-placed RnD front desk is a busy desk in ParaCode, unlike TG. 
I have no solution to offer here, I am simply addressing it as a constant that needs to be taken into consideration when it comes to porting this feature: there are no departmental lathes and in all likelihood never ever will be.
 
> All of your points are moot. This PR would work flawlessly for Paradise if not better since people work together on Paradise.
Nothing is flawless, otherwise AA (who is clearly heavily in favor of your PR) would not have marked it as a testmerge candidate, but I digress.
I dislike this tendency Paradise users have to portray TG as a sort of "Everything goes!" wasteland with no teamwork or rules. There is a lot that can be learned from their experience with features and their implementation. Otherwise neither you nor I would be defending their mechanics, even if our opinions differ on this exact implementation.


> **TL;DR** Massive station-wide threat? _TG Warden:_ Hey guys, I am not going to open the armoury because half of you will probably kill me on the spot and I only have six guns in here that actually fire lethals because cargo refuses to send me more guns- oh, and no one researched RND weapons because that requires toxins knowledge. _Para Warden:_ Wow someone did RND, hey HoS, tell the Captain to order lasers from Cargo, I'll go get kits from RND.
Read above.

Thank you for your hard work in porting this feature, and for taking the time to address the points raised in my first comment.

**Comment by Bm0n** [comment]
> Does not change the point that ordering from a readily available console in Brig, with a readily available budget (plus the Cargo chat console, I'd argue) makes it easier for Wardens get what they need and react quickly to the station needs, and even more so for simply just stocking up in advance: it's their budget, they use it as they see fit.

If you were an active TG player around the time parts kits were added you would know that those departmental cargo consoles didn't exist when parts kits were first added. The consoles and sec department budget are a relatively new addition to TG code.

You would also know that absolutely no one uses the cargo console IRC chat. This is true for both TG MRP and TG LRP. However, Paradise does have in its place the has wall-mounted request consoles that every department gets which serve the same purpose.

>Cargo works closer in tandem with other departments on TG than it does here on Para, exactly because of the points you raised: the fact that points are hard to get, and that the bounty list that Cargo gets always includes stuff from other departments, in turn actively forcing the Cargonians to scout out the other departments and cooperate with them if they want to order any of the memestuff they want, this encourages them to be proactive in the services they provide.

My man, I am sorry if this comes off as rude but you have only 46 rounds played on TG, most of them in observer, I have 1588 rounds. I am ranked 239 for overall hours on TG. This isn't true. At all. Fullstop.

You seemingly don't understand how modern bounties work on TG code so let me explain. Anyone and I mean anyone can go up to a public bounty console in front of cargo put their ID into that console and get a random bounty which will task them to get some item related to their job. Once they get whatever that bounty tasked them to get they can redeem it in front of cargo at that same public bounty console and where they'd get a bounty cube and then put that cube on the cargo desk, it is then at that point someone from cargo would place the bounty cube on the cargo shuttle. Notice how there is no interaction between cargo and the bounty doer until the bounty is complete. The only thing cargo needs to do for bounties is place the completed bounty cubes onto the cargo shuttle.

Yes, cargo can get their own bounties at the console but they'll only get tasks related to cargo which they can do themselves.

>As for QMs being accountable, you cannot hold accountable an inexperienced player, and inexperinced players are the norm for Cargo on Para. QMs are not heads and thus the server rules regarding heads and their competence, or the hour requirement both do not apply. More recent example: Cargo still orders riot shotguns against threats that require lethal force when Gamma isn't even in effect.

I would take any Paradise baldie over a TG QM any day of the week. Again, this comes down to knowing TG and how they behave over there. If you played a bit more TG you'd understand.

On Paradise cargo will get bwoinked if they start ordering tons of guns for no reason, this isn't true for TG. Most times TG cargo is more of an antagonistic threat than an ally even if there aren't any actual antags in cargo...

>Fun fact, I took this screenshot about 20 minutes in on my own personal server, and that includes the time it took for me to set up the SM, and the scrubber CO2 atmos pipenet for the coolers in the ordnance lab to work, since I am unfamiliar with the TG admin tools.
I could have even just published a Tier 1 "High Yield Explosives" experiment, which can be done using almost exactly the same steps the simplistic Toxins in Paradise has, but I actually took the time to make a Tritium/O2 mix for a Tier 3 paper instead.
This is literally just one person doing Toxins and then leaving the billion inexperienced Scientists to press buttons in front of a screen.
Or alternatively, doing it like a passive side job that does not require high engagement, while the scientist is busy with other stuff.

I am glad you were able to work out toxins on your own, not everyone can do that. I had to teach quite a few people it on TG because they simply couldn't wrap their heads around it on their own.

Still, no one ever prioritizes tech points on weapons development over T4 parts or other useful things. Don't kid yourself, it's much faster to get weapons tech up on Paradise.

>And yet, RnD runs out of materials all the time, all it takes is for ONE mineral to be in shortage in the lathe to require you to run the entire station's length for some Titanium, Silver, Uranium, or some other silly metal you never thought you'd reasonably run out of, this risk of running out is simply not present for TG players.

Kind of out of scope because this happens in current-day paradise regardless of parts kits. Go PR the ore silo, it wouldn't be hard to do.

>Before this PR, the standard procedure was to send Armory access, either in-person, or on a timed Guest Pass, officers would come to RnD and do all their shopping there (i.e. "can I have an AEG, an Xray, your best cell, a teleshield and NV SecHuds?")
Another thing this common technique granted was that RnD as well as Cargo and Sec can now arm people in case of a stationwide threat like a Blob or Terrors, the aforementioned distance between these three points actually works in favor of the crew since now they are always close to some part of the station that can offer them a chance at defending their lives.

Yeah, no more free guns from sci, that's kind of the whole point of this PR. You're going to have to get them from cargo or the armoury now. I see this as a good chance because of how frequently the protolathe is abused to spam out guns en masse for both traitors and security.

I actually predict that blob will die often now since people be forced to use welders and attack in waves instead of firing an AEG into the blob and running away which I see a lot of. I could come back to eat those words but we'll see.

>Not asking you to change that, I know heads are against it, that was one point I wanted to raise concerning the new TG-ported Cerberon layout, is that it does not take the fact that the awkwardly-placed RnD front desk is a busy desk in ParaCode, unlike TG.

Meta feedback goes to the Meta PR. This isn't the time or place to talk about it.

>Nothing is flawless, otherwise AA (who is clearly heavily in favor of your PR) would not have marked it as a testmerge candidate, but I digress.

Okay, maybe flawless is a little bit extreme word to use but still, I believe this will work better on Paradise than it ever did on TG. Sure as hell will be better than just allowing sci to produce as many guns as they want to.

>I dislike this tendency Paradise users have to portray TG as a sort of "Everything goes!" wasteland with no teamwork or rules. There is a lot that can be learned from their experience with features and their implementation. Otherwise neither you nor I would be defending their mechanics, even if our opinions differ on this exact implementation.

Some of it is true some of it isn't. I speak from my own experiences on TG and being the Paradise player with the most hours on TG besides trexter or imsxz, both of which don't play Para anymore, I feel like I have earned that right.

You're right in saying it's a not complete wasteland but at the same time it can be a complete hellhole with little cooperation between players, especially on LRP. A lot of their PRs are shaped around LRP with little consideration for TG MRP. Just how things are there. We have to pick and choose what we want from TG and what in turn works for Paradise.

**Comment by Xsingular** [comment]
> you have only 46 rounds played on TG
This can't be right, that number should be closer to 0 for this bkey/IP. The number 46 is pretty surprising to me.
I can only think of 3 or 4 times maximum where I slipped up.

Regardless, I definitely do **not** have as many hours on TG as you say you do (or its downstreams), and as a never-antag Science main, I welcome any change that makes RnD less of an antag heaven, I'm just putting a damper on the hype and voicing the concerns of those who don't check github as often as I do.

Good luck with the PR, and I can't wait to see the improvements it brings.

**Comment by JebusOurLordAndSaviour** [comment]
looks like a bad idea. games are supposed to be fun and not needing a degree in some made up science field to play, too complicated and stop trying to reinvent the wheel

**Comment by Bm0n** [comment]
> looks like a bad idea. games are supposed to be fun and not needing a degree in some made up science field to play, too complicated and stop trying to reinvent the wheel

You are playing Space Station 13, a game notorious for being overly in-depth and complex.

I don't think making it so that you have to craft advanced science guns makes an already very complex game overly complex. Please actually try to provide some constructive criticism next time instead of saying I am trying to reinvent the wheel.

File: code/modules/research/designs/weapon_designs.dm [hunk 1]
```diff
@@ -3,37 +3,33 @@
 /////////////////////////////////////////
 
 /datum/design/nuclear_gun
-	name = "Advanced Energy Gun"
-	desc = "An energy gun with an experimental miniaturized reactor."
+	name = "Advanced Energy Gun Parts Kit"
```
Comment (lewcc) [inline code/modules/research/designs/weapon_designs.dm:6]: I'm not entirely sure but I *think* you might want to add some \impropers here and there so it says 
"That's an Advanced Energy Gun Parts Kit"
```suggestion
	name = "\improper Advanced Energy Gun Parts Kit"
```

File: code/modules/research/designs/weapon_designs.dm [hunk 2]
```diff
@@ -3,37 +3,33 @@
...
+	desc = "The kit for an energy gun with an experimental miniaturized reactor."
```
Comment (lewcc) [inline code/modules/research/designs/weapon_designs.dm:7]: I feel like this works better here, though take this with a grain of salt!
```suggestion
	desc = "A kit for an energy gun with an experimental miniaturized reactor."
```

File: code/modules/research/designs/weapon_designs.dm [hunk 3]
```diff
@@ -48,14 +44,13 @@
 	category = list("Weapons")
 
 /datum/design/ioncarbine
-	name = "Ion Carbine"
+	name = "Ion Carbine Parts Kit"
 	desc = "How to dismantle a cyborg : The gun."
```
Comment (lewcc) [inline code/modules/research/designs/weapon_designs.dm:48]:
```suggestion
	desc = "How to dismantle a cyborg: The gun."
```

File: code/modules/research/designs/weapon_designs.dm [hunk 4]
```diff
@@ -121,25 +116,23 @@
...
-	build_path = /obj/item/gun/energy/lasercannon
-	locked = 1
+	build_path = /obj/item/weaponcrafting/gunkit/accelerator
 	category = list("Weapons")
 
 /datum/design/plasmapistol
-	name = "Plasma Pistol"
-	desc = "A specialized firearm designed to fire heated bolts of plasma. Can be charged up for a shield breaking shot."
+	name = "Plasma Pistol Parts Kit"
+	desc = "A specialized firearm kit designed to fire heated bolts of plasma. Can be charged up for a shield breaking shot."
```
Comment (lewcc) [inline code/modules/research/designs/weapon_designs.dm:130]:
```suggestion
	desc = "A kit for a specialized firearm designed to fire heated bolts of plasma. Can be charged up for a shield breaking shot."
```

File: code/modules/research/designs/weapon_designs.dm [hunk 5]
```diff
@@ -237,25 +229,23 @@
 	category = list("Weapons")
 
 /datum/design/xray
-	name = "Xray Laser Gun"
+	name = "Xray Laser Gun Parts Kit"
 	desc = "Not quite as menacing as it sounds"
```
Comment (lewcc) [inline code/modules/research/designs/weapon_designs.dm:233]: Might want to update this desc to note that it's a kit, as well as what the gun actually does!

**Review by lewcc — COMMENTED** [review]
Just some little nits, looks pretty good in general

File: code/modules/research/designs/weapon_designs.dm (hunk 1 above)
Comment (Bm0n) [reply code/modules/research/designs/weapon_designs.dm:6]: needs to be set in the guncrafting.dm file since that's where the actual parts kits items are. These are just the recipes you see on the RND console. Besides that everything else should be applied

**Comment by Inadvizable** [comment]
So far, from what I have seen of this (admittedly not that much yet), it is a very nice PR. Gun spam has always been a science problem and this also makes Cargo not completely fall off in favor of the aforementioned department.

**Comment by Norwes1** [comment]
Review as a Game Admin overseeing rounds as a whole:

...I love it. Seriously, it's wonderful. Cargo stays relevant, Science remains powerful but not "lol fuk all u's the moment we get uranium" sort of gamey, the Armoury is actually worth something, etc etc etc. I'm still holding out hope for the ore silo and departmental lathes, but I do really like this change for the meta as a whole.

Thoughts regarding the meta as a whole:
1. Midround blob/terror spiders: Heavily affected in favor of antags (no surprise). Blob in particular has a much easier go of things without immediate X-Ray spam, and terrors don't have to beeline straight for R&D, though they usually do anyway.
2. Nuke ops/wizard: Moderately affected. More exotic gear like plasma pistols generally didn't see much use, so this mostly means a couple fewer AEGs during war-ops.
3. Traitor, changeling, vampire. Heavily affected (against /certain/ antags). The powergamey types still gravitate to Science, but there's a lot fewer people running around with a dozen AEGs in their pack, which IMO is frankly a good thing. We don't need even more powercreep in this game, and there's still plenty of options for a smart Scientist (re: grenades of all shapes and sizes, lol) to use and abuse.
4. Cult: Moderately affected in favor of antags. Cult balance is A Problem, but that's an issue for a separate thread. This merge has mostly just cut down on the number of people on either side running around with overwhelming firepower, which is a good thing for game balance as a whole. Neither cult nor Seccies really needed to have umpty-bajillion AEGs fucking with 'em.
5. Minor antags (slaughter demon, revenant, etc.): Moderately affected in favor of antags. Definitely cuts down on the times some demon shows up to get AEG'd in the face by an unassuming Scientist with a Bag of Holding.

Overall, I really appreciate the way it's leveled the playing field a bit more. Teamwork can still carry the day, but now you need multiple departments cooperating to get the gud shit, rather than Science and a couple Miners basically making the rest of the station irrelevant.

File: code/modules/crafting/guncrafting.dm [hunk 6]
```diff
@@ -15,6 +15,65 @@
 	icon = 'icons/obj/improvised.dmi'
 	icon_state = "riflestock"
 
+///These gun kits are printed from the protolathe to then be used in making new weapons
+
+// GUN PART KIT //
+
+/obj/item/weaponcrafting/gunkit/
```
Comment (hal9000PR) [inline code/modules/crafting/guncrafting.dm:22]:
```suggestion
/obj/item/weaponcrafting/gunkit
```
tiny nitpick

**Review by hal9000PR — APPROVED** [review]
LGTM other than this. 

Quite a bit of dupe code, but I can see why you did it this way with the way crafting datums work.

File: code/modules/crafting/recipes.dm [hunk 7]
```diff
@@ -66,6 +66,158 @@
...
+	time = 20 SECONDS
+	category = CAT_WEAPONRY
+	subcategory = CAT_WEAPON
+
+/datum/crafting_recipe/tempgun/New()
+	..()
+	blacklist += subtypesof(/obj/item/gun/energy/gun)
+
+/datum/crafting_recipe/decloner
+	name = "Biological Demolecularisor"
```
Comment (lewcc) [inline code/modules/crafting/recipes.dm:100]: It's a little confusing that we call it decloner in the kit description, but call it this here. I'm not sure what the gun itself gets called by code (since I've honestly never seen it in game) but it might be good to standardize it.

File: code/modules/research/designs/weapon_designs.dm [hunk 8]
```diff
@@ -186,24 +179,23 @@
 	category = list("Weapons")
 
 /datum/design/stunrevolver
-	name = "Tesla Revolver"
-	desc = "A high-tech revolver that fires internal, reusable shock cartridges in a revolving cylinder. The cartridges can be recharged using conventional rechargers."
+	name = "Tesla Revolver Parts Kit"
+	desc = "The kit for a high-tech revolver that fires internal, reusable shock cartridges in a revolving cylinder. The cartridges can be recharged using conventional rechargers."
```
Comment (lewcc) [inline code/modules/research/designs/weapon_designs.dm:183]: Just so it's more or less standardized

```suggestion
	desc = "A kit for a high-tech revolver that fires internal, reusable shock cartridges in a revolving cylinder. The cartridges can be recharged using conventional rechargers."
```

File: code/modules/research/designs/weapon_designs.dm [hunk 9]
```diff
@@ -3,37 +3,33 @@
...
 	materials = list(MAT_METAL = 10000, MAT_GLASS = 2000, MAT_URANIUM = 3000, MAT_TITANIUM = 1000)
-	build_path = /obj/item/gun/energy/gun/nuclear
-	locked = 1
+	build_path = /obj/item/weaponcrafting/gunkit/nuclear
 	category = list("Weapons")
 
 /datum/design/decloner
-	name = "Decloner"
+	name = "Decloner Parts Kit"
 	desc = "Your opponent will bubble into a messy pile of goop."
```
Comment (lewcc) [inline code/modules/research/designs/weapon_designs.dm:17]: I love this description but you say in all the others that it's "a kit for...", you might just want to standardize this

**Review by lewcc — COMMENTED** [review]
Some small consistency things on the messages

File: code/modules/crafting/recipes.dm (hunk 7 above)
Comment (Bm0n) [reply code/modules/crafting/recipes.dm:100]: The real name is biological demolecularisor, decloner is more short form. It was called decloner in protolathe so I left it as is(probably something to do with the protolathes UI but not sure)

File: code/modules/research/designs/weapon_designs.dm (hunk 9 above)
Comment (Bm0n) [reply code/modules/research/designs/weapon_designs.dm:17]: This isn't the actual items parts kit description, just a short blob of text that you see in the protolathes UI before printing the kit. I think it's fine so long as it has parts kit in the actual name.
